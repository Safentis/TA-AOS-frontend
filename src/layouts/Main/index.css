.main {
  background-color: var(--main-grey-color);
}
