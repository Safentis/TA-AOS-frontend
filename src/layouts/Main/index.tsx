import React, { FC } from 'react';

import './index.css';
import { Props } from './index.interface';

const Main: FC<Props> = ({
  children,
  className = '',
}): React.ReactElement => {
  return (
    <main className={`main ${className}`}>
      <div className="main__inner">{children}</div>
    </main>
  );
};

export default Main;
