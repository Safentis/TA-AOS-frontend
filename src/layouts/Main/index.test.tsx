import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import Main from './index';

import { Props } from './index.interface';

const setUp = (props: Props) => {
  return shallow(<Main {...props} />);
};

describe('<Main />', () => {
  let component: ShallowWrapper<
    any,
    Readonly<{}>,
    React.Component<{}, {}, any>
  >;
  let props: Props = {
    className: 'custom-style',
  };

  beforeEach(() => {
    component = setUp(props);
  });

  it('component does match to the snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
