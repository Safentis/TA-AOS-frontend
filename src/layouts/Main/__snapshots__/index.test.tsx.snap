// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> component does match to the snapshot 1`] = `
<main
  className="main custom-style"
>
  <div
    className="main__inner"
  />
</main>
`;
